Cabal/Distribution/Simple/Haddock.hs

```diff
@@ -276,23 +275,31 @@ componentGhcOptions verbosity lbi bi clbi odir =
                        "haddock only supports GHC and GHCJS"
   in f verbosity lbi bi clbi odir
 
-fromLibrary :: Verbosity
-            -> FilePath
-            -> LocalBuildInfo -> Library -> ComponentLocalBuildInfo
-            -> Maybe PathTemplate -- ^ template for HTML location
-            -> Version
-            -> IO HaddockArgs
-fromLibrary verbosity tmp lbi lib clbi htmlTemplate haddockVersion = do
-    inFiles <- map snd `fmap` getLibSourceFiles lbi lib clbi
+type ModuleFiles = [(ModuleName.ModuleName, FilePath)]
+fromTarget :: (LocalBuildInfo -> ComponentLocalBuildInfo
+                -> target -> IO ModuleFiles)
```

Comment:
Yes, it was of typing at the use site... It's a subjective trade-off, but I went for compactness.